packages/server/src/sdk/app/views/utils.ts

```diff
@@ -0,0 +1,39 @@
+import { ViewV2 } from "@budibase/types"
+import { utils, dataFilters } from "@budibase/shared-core"
+import { isPlainObject } from "lodash"
+import { HTTPError } from "@budibase/backend-core"
+
+function isEmptyObject(obj: any) {
+  return obj && isPlainObject(obj) && Object.keys(obj).length === 0
+}
+
+export function ensureQueryUISet(view: ViewV2) {
+  if (!view.queryUI && view.query && !isEmptyObject(view.query)) {
```

Comment:
I understand this affects the input therefore it isn't required to return and maybe this is a personal preference, but even if the input is changed I still like when a function returns the altered state - therefore you know the function has a goal of changing/updating something, the returned response will be it.

It would also be possible to do a:
```
return {
  ...view,
  queryUI: utils.processSearchFilters(view.query),
}
```
To get a response which does not affect the input; I understand this is less performant and more GC time however. Having it just `return view` even if the input has changed is helpful to the caller, since it makes it a bit more obvious what it did.

Reply:
I'd argue the opposite, it is confusing to the caller when a function both modifies its input and returns it, and we've had this bite us recently with `array.sort()`.

I think functions should do one or the other, and if there's no clear need to copy I think mutating makes the most sense.

Reply:
In this case I think I would probably prefer the copy, as the function is called quite a bit and its not very obvious if its just error checking. I do also like having the response since it makes it properly chainable e.g. `checkOtherViewParameters(ensureQueryUISet(...))` - although I guess this is still achievable with in/out parameters.

Middle-ground could be a name which explains it mutates the input, just so its obvious what it will do.

Reply:
I feel like this:

```js
ensureQueryUISet(view)
checkOtherViewParameters(view)
```

Is clearer than:

```js
checkOtherViewParameters(ensureQueryUISet(view))
```

But if your preference is copying I'll change it.